src/main/java/com/hazelcast/aws/TagFilters.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.hazelcast.aws;
+
+import com.hazelcast.aws.Tags.Tag;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import static com.hazelcast.aws.StringUtils.isNotEmpty;
+
+final class TagFilters {
```

Comment:
This is just a util class with one method. I think you can move (and rename) this method to `AwsEc2Api`, since it's only used there. Or you can move it to the `Filter` class if you prefer.

Reply:
I have moved this functionality back to  `AwsEc2Api` as suggested although I found it a little bit harder to test if these tag specific filters are built correctly. I'll leave it for you to judge and if needed I can still do some adjustments.